### DISCUSSION TITLE
How can I add a sidebar to the blog-start layout?

### DISCUSSION THREAD

**Discussion by arkholt** [datenstrom/community#1002] [Ask a question] [answered]
I haven't updated my yellow blog since version 0.6, so I'm just starting over with a fresh version 0.9 installation. The only issue I've run into so far is with the theme. My theme has a sidebar, and I'm able to add the php to make it appear on the default layout, but whenever I try to add it to the blog-start layout, it says "file not found" and doesn't display anything else, or even the sidebar. I have the sidebar.md in the shared folder, with Status: shared in the yaml frontmatter.

My default sidebar layout looks like this:

```
<?php $this->yellow->layout("header") ?>
<div class="<?php echo "content".($this->yellow->page->isPage("sidebar") ? " with-sidebar" : "") ?>">
<div class="sidebar" role="complementary">
<?php echo $this->yellow->page->getPage("sidebar")->getContentHtml() ?>
</div>
<div class="main" role="main">
<h1><?php echo $this->yellow->page->getHtml("titleContent") ?></h1>
<?php echo $this->yellow->page->getContentHtml() ?>
</div>
</div>
<?php $this->yellow->layout("footer") ?>
```

And it works perfectly fine. Where should I add the php code to the blog-start layout for it to work?

EDIT: I was able to add

```
<div class="<?php echo "content".($this->yellow->page->isPage("sidebar") ? " with-sidebar" : "") ?>">
<div class="sidebar" role="complementary">
<?php echo $this->yellow->page->getPage("sidebar")->getContentHtml() ?>
</div>
```

...to the blog entry layout while replacing

```
<div class="content">
```

...and did not have any issues. For some reason the only layout I'm not able to add this to is the blog-start layout. The error I get is "File not found
The requested file was not found. Oh no..."

**Comment by annaesvensson** [comment, marked as answer]
From what I can see, you have done everything correct.

1. Add the class `with-sidebar` for your custom CSS.
2. Add the `<div>...</div>` with your sidebar HTML.
3. Repeat the same steps with each layout file you want to change.

The error message tells me that something else has gone wrong. 

Tip: The [blog extension](https://github.com/annaesvensson/yellow-blog) shows the error message `File not found` if blog pages can not be found. You mentioned that you recently updated Datenstrom Yellow and I suspect that the cause of the problem lies here. Remember that in Datenstrom Yellow 0.8.2 the page setting `Template: blog` has been changed to `Layout: blog`. You probably have to adjust some [page settings](https://datenstrom.se/yellow/help/how-to-change-the-system#page-settings), have a look inside your content folder. See also discussions datenstrom/community#403 and datenstrom/community#438.

**Reply by arkholt** [reply]
Ah, I see. Thank you for your reply.

What I had done was duplicate the blog layout and the blog-start layout, and add the sidebar code to the new ones, and named them blog-sidebar and blog-start-sidebar. For some reason when I did that, it said "file not found," even when I put the layout as "blog-sidebar" in the yaml front matter of the blog post. If I add it directly to blog-start and blog layouts, it works fine. I didn't want to edit the existing layouts out of caution, but I guess I will just do that in the future, as it doesn't seem to like it when I try to add new ones.

**Reply by annaesvensson** [reply]
Good idea to have separate files for custom HTML and CSS.

I don't know if you already knew that themes can have their own layout files, this is a core feature, maybe this is something you want to try out. For example the file `system/layouts/blog-start.html` will be used with any theme, the file `system/layouts/arkholt-blog-start.html` will only be used with `Theme: arkholt`. 

For clarification, the blog start page must have the setting `Layout: blog-start` and individual blog pages must have the setting `Layout: blog`. This can not be changed. The blog extension source code specifically looks for these page settings in content files, it is a kind of identifier that tells the blog extension what type of page a content file is.

**Reply by arkholt** [reply]
Ah, now I see what my error was. I wasn't aware of that naming scheme for the layouts. Thank you for the clarification. Everything appears to be working now.
